app/src/main/java/cm/aptoide/pt/promotions/ClaimPromotionDialogFragment.java

```diff
@@ -340,18 +342,32 @@ public void onResume() {
     dismiss();
   }
 
-  private void handleClipboardPaste() {
-    String address;
+  @Override public void fetchWalletAddressByIntent() {
+    if (walletErrorView.getVisibility() != View.VISIBLE) {
+      ((ActivityResultNavigator) getContext()).navigateForResult(Intent.ACTION_VIEW,
+          Uri.parse(WALLET_INTENT_URI_ACTION), WALLET_INTENT_REQUEST_CODE, WALLET_INTENT_EXTRA_KEY,
+          WALLET_INTENT_EXTRA_VALUE);
```

Comment:
This logic should be in a ClaimPromotionNavigator and be called by the presenter. The strings should then be passed in the navigator. (Check AppNavigator)